libs/model/src/services/evmChainEvents/chain-event-utils.ts

```diff
@@ -291,6 +292,37 @@ const xpChainEventCreatedMapper: EvmMapper<'XpChainEventCreated'> = (
   };
 };
 
+const transferSingleMapper: EvmMapper<'NamespaceTransferSingle'> = (
+  event: EvmEvent,
+) => {
+  const decoded = decodeLog({
+    abi: erc1155Abi,
+    eventName: 'TransferSingle',
+    data: event.rawLog.data,
+    topics: event.rawLog.topics,
+  });
+
+  const args = decoded.args as Record<string, unknown>;
```

Comment:
The point of Viem is to strictly type decoded events like this so type coercing `decoded.args` to an arbitrary record is incorrect.

Why did you do this? Type seems to work fine for me:

![image](https://github.com/user-attachments/assets/8d69f8e0-0f87-4433-8412-6935ac374589).

You shouldn't have to do all the type coercions you do below either.